ui/prometheus-plugin/src/plugins/prometheus-variables.tsx

```diff
@@ -110,3 +161,29 @@ export const PrometheusLabelValuesVariable: VariablePlugin<PrometheusLabelValues
   OptionsEditorComponent: PrometheusLabelValuesVariableEditor,
   createInitialOptions: () => ({ label_name: '' }),
 };
+
+export const PrometheusPromQLVariable: VariablePlugin<PrometheusPromQLVariableOptions> = {
+  getVariableOptions: async (spec, ctx) => {
+    const client: PrometheusClient = await ctx.datasourceStore.getDatasourceClient(spec.datasource ?? DEFAULT_PROM);
+    // TODO we may want to manage a range query as well.
```

Comment:
I think it's ok to target `/api/v1/query` only, at least that's what grafana is doing with `query_result()`. For volatile stuff you can just use an <aggr>_over_time function in your instant query (like avg_over_time(up[1h])), and this should do the trick.

Reply:
Yeah maybe we can start simple then add it in.